### PR TITLE
Feature/external annotation processors

### PR DESCRIPTION
**Pull request by jaredlll08**
I believe I've done this correctly, it seems to be working fine anyway.

Some caveats:

* We have a bit of a stupid naming situation with `crafttweaker_annotations` package, not that we didn't have it before, but now it is in code we actually work in instead of a different project.
* Running `checkoutAP` doesn't refresh idea, but it should only really be done when debugging the AP
* We will need to update the AP version like we do ZenCode

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -32,4 +32,5 @@ logs
 fabric-game-tests.html
 
 secrets.json
-changelog.md
\ No newline at end of file
+changelog.md
+CraftTweaker-Annotation-Processors
```
Comment (kindlich) [inline .gitignore]: Do we want a newline at EOF?

**Review by kindlich — COMMENTED** [review]
The only thing I'm a bit sceptical about is how to make sure both the AP stub classes and the classes here are both modified at once instead of only in one place and having them become different over time?

But since those Annotations probably won't be touched too much, it should be fine?

**Comment by jaredlll08** [comment]
> The only thing I'm a bit sceptical about is how to make sure both the AP stub classes and the classes here are both modified at once instead of only in one place and having them become different over time?
> 
> But since those Annotations probably won't be touched too much, it should be fine?

That was actually an issue we had already in a way:
https://github.com/CraftTweaker/CraftTweaker/blob/55453b3220e8d4622762eea62d4ff94f93bc7c5f/Annotations/Crafttweaker_Annotation_Processors/src/main/java/com/blamejared/crafttweaker_annotation_processors/processors/validation/BracketHandlerCheckValidationProcessor.java#L20-L21
https://github.com/CraftTweaker/CraftTweaker/blob/55453b3220e8d4622762eea62d4ff94f93bc7c5f/Annotations/Crafttweaker_Annotation_Processors/src/main/java/com/blamejared/crafttweaker_annotation_processors/processors/validation/PreprocessorAnnotationValidationProcessor.java#L17-L21
We were referencing classes as strings so they were incorrect when we changed the package name.

But yes, we will need to ensure that any changes are reflected in the stub.

File: build.gradle.kts [hunk 2]
```diff
@@ -290,3 +287,13 @@ tasks.create("postDiscord") {
...
+
+tasks.create("checkoutAP") {
+    if (!rootProject.file("CraftTweaker-Annotation-Processors").exists()) {
+        exec {
+            commandLine("git", "clone", "https://github.com/CraftTweaker/CraftTweaker-Annotation-Processors.git")
+        }
+    } else {
+        println("CraftTweaker-Annotation-Processors folder already exists!")
+    }
+}
```
Comment (TheSilkMiner) [inline build.gradle.kts]: Do we also want a `uncheckoutAP` task?

File: build.gradle.kts [hunk 3]
```diff
@@ -290,3 +287,13 @@ tasks.create("postDiscord") {
     }
 
 }
+
+tasks.create("checkoutAP") {
+    if (!rootProject.file("CraftTweaker-Annotation-Processors").exists()) {
```
Comment (TheSilkMiner) [inline build.gradle.kts]: Maybe checking if it has contents rather than if the directory exists?

File: settings.gradle.kts [hunk 4]
```diff
@@ -33,26 +33,33 @@ pluginManagement {
...
 
 include("ZenCode")
 collectSubProjects("ZenCode")
-collectSubProjects("Annotations")
 
 include("Common")
 include("Fabric")
 include("Forge")
 
+if (File(rootDir, "CraftTweaker-Annotation-Processors").exists()) {
```
Comment (TheSilkMiner) [inline settings.gradle.kts]: Remind me, does `file` not exist here? Or cannot we resolve stuff from `rootDir`?

**Review by TheSilkMiner — APPROVED** [review]

**Pull request merged**